test/libcommonserver/log/testlog.cpp

```diff
@@ -45,4 +52,257 @@ void TestLog::testLog() {
+        bool tooDeep = false;
+        IoHelper::getDirectorySize(tempDir.path, tempDirSize, err, 0);
+        CPPUNIT_ASSERT(err == IoErrorSuccess || err == IoErrorMaxDepthExceeded);
+        CPPUNIT_ASSERT_EQUAL(parmsDbSize, tempDirSize);
+    }
+}
+
+void TestLog::testCompressLogs(void) {
+    {
+        TemporaryDirectory tempDir;
+
+        std::ofstream logFile(tempDir.path / "test.log");
+        for (int i = 0; i < 10000; i++) {
+            logFile << "Test log line " << i << std::endl;
+        }
+        logFile.close();
+
+        const SyncPath logDir = tempDir.path / "log";
+        IoError err = IoErrorSuccess;
+        CPPUNIT_ASSERT_EQUAL(true, IoHelper::createDirectory(logDir, err));
+        CPPUNIT_ASSERT_EQUAL(IoErrorSuccess, err);
+
+        const SyncPath logFilePath = logDir / "test.log";
+        CPPUNIT_ASSERT_EQUAL(true, IoHelper::copyFileOrDirectory(tempDir.path / "test.log", logFilePath, err));
+        CPPUNIT_ASSERT_EQUAL(IoErrorSuccess, err);
+
+        uint64_t logDirSize = 0;
+        bool tooDeep = false;
+
+        CPPUNIT_ASSERT_EQUAL(true, IoHelper::getDirectorySize(logDir, logDirSize, err, 0));
+        CPPUNIT_ASSERT(err == IoErrorSuccess || err == IoErrorMaxDepthExceeded);
+        CPPUNIT_ASSERT(logDirSize >= 0);
+
+        ExitCause cause = ExitCauseUnknown;
+        ExitCode exitCode = Log::instance()->compressLogFiles(tempDir.path, cause);
+
+        CPPUNIT_ASSERT_EQUAL(ExitCauseUnknown, cause);
+        CPPUNIT_ASSERT_EQUAL(ExitCodeOk, exitCode);
+
+        uint64_t tempDirSize = 0;
+        IoHelper::getDirectorySize(tempDir.path, tempDirSize, err, 0);
+        CPPUNIT_ASSERT(err == IoErrorSuccess || err == IoErrorMaxDepthExceeded);
+        CPPUNIT_ASSERT(tempDirSize < logDirSize);
+
+        bool exists = false;
+        CPPUNIT_ASSERT_EQUAL(true, IoHelper::checkIfPathExists(tempDir.path / "test.log.gz", exists, err));
+        CPPUNIT_ASSERT_EQUAL(IoErrorSuccess, err);
+        CPPUNIT_ASSERT_EQUAL(true, exists);
+
+        CPPUNIT_ASSERT_EQUAL(true, IoHelper::checkIfPathExists(logDir / "test.log.gz", exists, err));
+        CPPUNIT_ASSERT_EQUAL(IoErrorSuccess, err);
+        CPPUNIT_ASSERT_EQUAL(true, exists);
+    }
+
+    {  // test the progress bar
+        TemporaryDirectory tempDir;
+        for (int i = 0; i < 30; i++) {
+            std::ofstream logFile(tempDir.path / ("test" + std::to_string(i) + ".log"));
+            for (int j = 0; j < 10; j++) {
+                logFile << "Test log line " << j << std::endl;
+            }
+            logFile.close();
+        }
+
+        int percent = 0;
+        int oldPercent = 0;
+        std::function<void(int)> progress = [&percent, &oldPercent](int p) {
+            percent = p;
+            CPPUNIT_ASSERT(percent >= oldPercent);
+            oldPercent = percent;
+        };
+
+        ExitCause cause = ExitCauseUnknown;
+
+        ExitCode exitCode = Log::instance()->compressLogFiles(tempDir.path, cause, progress);
```

Comment:
```suggestion
        const ExitCode exitCode = Log::instance()->compressLogFiles(tempDir.path, cause, progress);
```

Reply:
Done in commit: https://github.com/Infomaniak/desktop-kDrive/pull/100/commits/8b66441e49e63e5dede7a1bac5a21b1bc33cc8a0